client/src/lib/supabase/middleware.ts

```diff
@@ -78,60 +58,55 @@ export async function updateSession(request: NextRequest) {
 
   //simple pathname variable
   const pathname = request.nextUrl.pathname;
-  console.log("USER REQUEST PATHNAME: "+ pathname);
   
   //Auth user
   if (user) {
     const userEmail = user?.email;
     console.log("USER " + userEmail + " HAS AUTHENTICATED");
     
-    //Task 1: Only allow authorized users to access the pages under the (main) directory using whitelist strategy
+    // redirect authenticated users from home, login, and signup to explore
+    if (pathname === "/" || pathname === "/signup" || pathname === "/login") {
+      const url = request.nextUrl.clone();
+      url.pathname = "/explore";
+      return NextResponse.redirect(url);
+    }
+
+    // Only allow authorized users to access the pages under the (main) directory using whitelist strategy
     if(!(isAllowedUserPath(pathname)) && !(isPublicPath(pathname))) {
-      console.log("User attempted to access a restricted path. Redirecting to /404");
-      console.log("Task 1: Only allow authorized users to access the pages under the (main) directory");
       const url = request.nextUrl.clone();
-      if (url.pathname.indexOf("/404") < 0) {
-        url.pathname = "/404";
+      if (url.pathname.indexOf("/401") < 0) {
```

Comment:
what does this do? I know you tried explaining it but I never understood this

Reply:
This ensures that when we redirect to 404, it does not continuously loop on the redirect